data_processing_scripts/fastq_script.py

```diff
@@ -0,0 +1,154 @@
+# Import dna_rna_dict.py containing dictionaries for working with dna and rna sequences
+import dna_rna_dict as drd
+import os
+
+# GC content frequency function
+
+def gc_content(sequence: str) -> float:
+    """
+    Calculates the GC content in a fastq sequence.
+
+    :param sequence: fastq sequence
+    :type sequence: str
+    :return: GC content as a percentage
+    :rtype: float
+    """
+    gc_count = sequence.count('G') + sequence.count('C')
+    total_count = len(sequence)
+    gc_content1 = round((gc_count / total_count) * 100, 2)
+    return gc_content1
+
+
+# fastq sequence length function
+
+
+def seq_length(sequence: str) -> int:
+    """
+    Calculates the length of fastq sequence.
+
+    :param sequence: fastq sequence
+    :type sequence: str
+    :return: the length as a number
+    :rtype: int
+    """
+    return len(sequence)
+
+
+# the mean encoding offset for quality scores function
+
+
+def mean_encoding_offset(quality_string: str) -> float:
+    """
+    Calculate the mean encoding offset for quality scores in the input string.
+
+    :param quality_string: A string of quality scores
+    :type quality_string: str
+    :return: The mean encoding offset
+    :rtype: float
+    """
+    total_offset = 0
+    for char in quality_string:
+        offset = ord(char) - 33  # Assuming 33 as the default encoding offset
+        total_offset += offset
+    mean_offset = total_offset / len(quality_string)
+    return mean_offset
+
+
+def main_fastq_tools(seqs, gc_bounds=(0, 100), length_bounds=(0, 2 ** 32), quality_threshold=0):
```

Comment:
1) Ну это же не то чтобы tools, это фильтр. Лучше так конкретно и назыввать. 
2) Функция должна быть глаголом. Тут понятно что она главная конечно, но в ущерб пониманию того что она делает сама:
```suggestion
def run_fastq_filter(seqs, gc_bounds=(0, 100), length_bounds=(0, 2**32), quality_threshold=0):
```